### ISSUE TITLE
Occasional S3 "no such key" error messages.

### ISSUE THREAD

**Issue by hwine** [mozilla-services/fx-sig-verify#14]
reported by @Micheletto -- seeing occasional failures due to:
```
'NoSuchKey(u'An error occurred (NoSuchKey) when calling the GetObject operation: The specified key does not exist.
```

Examples: 
- [Firefox%20Setup%2054.0.1.exe](https://archive.mozilla.org/pub/firefox/candidates/54.0.1-candidates/build1/partner-repacks/funnelcake114/v3/win32/en-US/Firefox%20Setup%2054.0.1.exe)

**Comment by hwine** [comment]
Next steps:

- [ ] Check an instance to see if version number is > 1
- [ ] Script to check CloudTrail logs for these events
- [ ] Check into the "wait for" type api calls

**Comment by hwine** [comment]
For the previous 24 hours:
```
6587 function invocations
5960 pass
  34 fail for (legitimate) lack of signature
 593 fail for "NoSuchKey" error from S3 (~9%)
```

Trying a 'waiter' function for s3, if it fails first time.

**Comment by hwine** [comment]
@oremj noticed a pattern -- the keys that fail all contain '+' characters to represent spaces. AWS [cautions against](http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-key-guidelines-special-handling) using '+' in key names.

Workaround is to use ``urllib.unquote_plus(key_name)`` before submitting key.

**Comment by hwine** [comment]
The type of files that have the '``+``' in their names are only uploaded for Beta & GA releases. Waiting for one of those to happen to confirm fix.

**Comment by hwine** [comment]
fixed in 7ffb0d59, part of v0.3.0

This code still includes debugging code to ensure this is the problem. Cleanup elsewhere.
